feature-libs/my-account/organization/src/components/budget/edit/budget-edit.component.ts

```diff
@@ -59,10 +53,7 @@ export class BudgetEditComponent {
       form.disable();
       this.budgetService.update(budgetCode, form.value);
 
-      this.routingService.go({
-        cxRoute: 'budgetDetails',
-        params: form.value,
-      });
+      this.currentBudgetService.launch('budgetDetails', form.value);
```

Comment:
I can understand introducing `launch` method saves us one constructor parameter (RoutingService). Maybe you can help me learn some more advantages of introducing it? 
(Perhaps you have some specific customization scenario in mind - i.e. customer overwrites this method to launch in modal?)

That said, I can see many arguments why not to put `launch` method in all `CurrentXXXService`s:
- IMO it's not the business of `CurrentBudgetService` to perform routing navigations or launch any view. It's only an util service to expose the current model based on the application context (by coincidence, it's the routing context)
- nor the arguments of the `launch`method, neither the implementation, make any use of the _current budget_ model or code
- the signature of the method is a bit confusing - we pass an opinionated string (which happens to be a cxRoute name) and model of the form. I needed to take a look into the implementation of `launch` method to learn it's a Spartacus navigation
- if we used simply the `RoutingService` in the component, it would be very straightforward and intuitive to me
- 4 constructor dependencies of BudgetEditComponent (including Routing Service as the fourth one) wouldn't look too bad IMHO

Reply:
Agreed! I've left it out on this service as it's not his business.